File: API_REST/API_REST_DSW/src/java/Model/Usuario.java
package Model;

import java.io.Serializable;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

/**
 *
 * @author Ivens
 */
@Entity
@Table(uniqueConstraints={@UniqueConstraint(columnNames={"cpf"})})
public class Usuario implements Serializable{

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long ID;
    private String cpf;    
    private String nomeCompleto;    
    private String senha;
    private String email;
    private boolean administrador;
    @ManyToOne(cascade = CascadeType.ALL)
    private Endereco endereco;

    public Long getID() {
        return ID;
    }

    public void setID(Long ID) {
        this.ID = ID;
    }
    
    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getNomeCompleto() {
        return nomeCompleto;
    }

    public void setNomeCompleto(String nomeCompleto) {
        this.nomeCompleto = nomeCompleto;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public boolean isAdministrador() {
        return administrador;
    }

    public void setAdministrador(boolean administrador) {
        this.administrador = administrador;
    }

    public Endereco getEndereco() {
        return endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }
    
    
}

File: front-end-dsw/src/telas/Usuario.js

var UserProfile = (function() {
    var cpfUser = "";
    var carrinhoCompras = [];
    var admin = false;
    var nomeCompleto = "";
    var email = "";
    var endereco = "";

    var getNomeCompleto = function () {
        return nomeCompleto;
    };

    var getEmail = function () {
        return email;
    };

    var getEndereco = function () {
        return endereco;
    };

    var setNomeCompleto = function (nomeUser) {
        nomeCompleto = nomeUser;
    };

    var setEmail = function (emailUser) {
        email = emailUser;
    };

    var setEndereco = function (enderecoUser) {
        endereco = enderecoUser;
    }

    var getCpf = function() {
        return cpfUser;
    };

    var getCarrinho = function() {
        return carrinhoCompras;
    };

    var setCarrinho = function(carrinho) {
        carrinhoCompras = carrinho;
    }

    var setCpf = function(cpf) {
        cpfUser = cpf;
    };

    var setAdmin = function(userAdmin){
        admin = userAdmin;
    };

    var getAdmin = function(){
        return admin;
    }

    return {
        getCpf: getCpf,
        setCpf: setCpf,
        getCarrinho: getCarrinho,
        setCarrinho: setCarrinho,
        getAdmin: getAdmin,
        setAdmin: setAdmin,
        setEndereco: setEndereco,
        getEndereco: getEndereco,
        setEmail: setEmail,
        getEmail: getEmail,
        setNomeCompleto: setNomeCompleto,
        getNomeCompleto: getNomeCompleto
    }

})();


export default UserProfile;